plugins/rest/plugin_impl_rest.go

```diff
@@ -28,37 +28,50 @@ import (
 	"github.com/ligato/vpp-agent/plugins/vpp"
 	aclvppcalls "github.com/ligato/vpp-agent/plugins/vpp/aclplugin/vppcalls"
 	ifvppcalls "github.com/ligato/vpp-agent/plugins/vpp/ifplugin/vppcalls"
+	ipsecvppcalls "github.com/ligato/vpp-agent/plugins/vpp/ipsecplugin/vppcalls"
 	l2vppcalls "github.com/ligato/vpp-agent/plugins/vpp/l2plugin/vppcalls"
 	l3vppcalls "github.com/ligato/vpp-agent/plugins/vpp/l3plugin/vppcalls"
-)
-
-const (
-	swIndexVarName = "swindex"
+	l4vppcalls "github.com/ligato/vpp-agent/plugins/vpp/l4plugin/vppcalls"
 )
 
 // REST api methods
 const (
-	GET = "GET"
+	GET  = "GET"
+	POST = "POST"
 )
 
 // Plugin registers Rest Plugin
 type Plugin struct {
 	Deps
 
-	indexItems []indexItem
+	// Partial index items
+	aclIndexItems       []indexItem
+	ifIndexItems        []indexItem
+	ipSecIndexItems     []indexItem
+	l2IndexItems        []indexItem
+	l3IndexItems        []indexItem
+	l4IndexItems        []indexItem
+	telemetryIndexItems []indexItem
+	commonIndexItems    []indexItem
 
 	// Channels
 	vppChan  api.Channel
 	dumpChan api.Channel
 
 	// Handlers
-	aclHandler aclvppcalls.AclVppRead
-	ifHandler  ifvppcalls.IfVppRead
-	bfdHandler ifvppcalls.BfdVppRead
-	bdHandler  l2vppcalls.BridgeDomainVppRead
-	fibHandler l2vppcalls.FibVppRead
-	xcHandler  l2vppcalls.XConnectVppRead
-	rtHandler  l3vppcalls.RouteVppRead
+	aclHandler   aclvppcalls.AclVppRead
+	ifHandler    ifvppcalls.IfVppRead
+	bfdHandler   ifvppcalls.BfdVppRead
+	natHandler   ifvppcalls.NatVppRead
+	stnHandler   ifvppcalls.StnVppRead
+	ipSecHandler ipsecvppcalls.IPSecVPPRead
+	bdHandler    l2vppcalls.BridgeDomainVppRead
+	fibHandler   l2vppcalls.FibVppRead
+	xcHandler    l2vppcalls.XConnectVppRead
+	arpHandler   l3vppcalls.ArpVppRead
+	pArpHandler  l3vppcalls.ProxyArpVppRead
+	rtHandler    l3vppcalls.RouteVppRead
+	l4Handler    l4vppcalls.L4VppRead
 
 	sync.Mutex
```

Comment:
This should use much clearer name and not be embedded. Something like: `govppMutex`.